registry/middleware/repository/middleware.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"fmt"
+
+	"github.com/docker/distribution"
+)
+
+// InitFunc is the type of a RepositoryMiddleware factory function and is
+// used to register the contsructor for different RepositoryMiddleware backends.
```

Comment:
Might be interesting to note here that RepositoryMiddleware may be request scoped.